File: iosApp/iosApp/Views/Calendar/EpisodeCalendarItem.swift
//
//  EpisodeCalendarItem.swift
//  iosApp
//
//  Created by Owen LeJeune on 2026-02-09.
//

import SwiftUI
import Shared

struct EpisodeCalendarItem: View {
    let episodeGroup: EpisodeGroup
    
    private var episode: Episode {
        episodeGroup.first
    }
    
    private var isPremier: Bool {
        episode.seasonNumber == 1 && episode.episodeNumber == 1
    }
    
    private var statusIcon: String? {
        if episode.hasFile {
            return "checkmark.circle.fill"
        } else if !episode.monitored {
            return "bookmark"
        } else if !episode.hasAired {
            return "clock.fill"
        } else if episode.monitored {
            return "bookmark.fill"
        } else if !episode.monitored {
            return "bookmark"
        }
        return nil
    }
    
    private var airTime: String? {
        guard let airDateUtc = episode.airDateUtc else { return nil }
        
        let timeInterval = TimeInterval(airDateUtc.epochSeconds)
        let date = Date(timeIntervalSince1970: timeInterval)
        
        let formatter = DateFormatter()
        formatter.dateStyle = .none
        formatter.timeStyle = .short
        return formatter.string(from: date)
    }
    
    var body: some View {
        HStack(spacing: 12) {
            if let series = episode.series {
                PosterItem(item: series, radius: 8)
                    .frame(width: 45)
            }
            
            VStack(alignment: .leading, spacing: 2) {
                Text(episode.series?.title ?? MR.strings().unknown.localized())
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(.primary)
                    .lineLimit(1)
                
                Text("S\(episode.seasonNumber)E\(episode.episodeNumber) • \(episode.title ?? "")")
                    .font(.system(size: 14))
                    .foregroundColor(.secondary)
                    .lineLimit(1)
                
                HStack(spacing: 6) {
                    if let airTime = airTime {
                        Label(airTime, systemImage: "clock")
                            .font(.system(size: 12, weight: .medium))
                            .foregroundColor(.secondary)
                    }
                    
                    if isPremier {
                        BadgeView(text: MR.strings().premier.localized(), color: .orange)
                    }
                    
                    if let finaleType = episode.finaleType {
                        BadgeView(text: finaleType.resource.localized(), color: .purple)
                    }
                    
                    if !episodeGroup.additional.isEmpty {
                        Text("+\(episodeGroup.additional.count)")
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(.blue)
                    }
                }
                .padding(.top, 2)
            }
            
            Spacer()
            
            if let icon = statusIcon {
                Image(systemName: icon)
                    .font(.system(size: 20))
                    .foregroundColor(episode.hasFile ? .green : .secondary)
            }
        }
        .padding(10)
        .background(Color(.secondarySystemBackground))
        .cornerRadius(12)
    }
}

File: composeApp/src/androidMain/kotlin/com/dnfapps/arrmatey/ui/calendar/EpisodeCalendarItem.kt
package com.dnfapps.arrmatey.ui.calendar

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTimeFilled
import androidx.compose.material.icons.filled.Bookmark
import androidx.compose.material.icons.filled.BookmarkBorder
import androidx.compose.material.icons.filled.FileDownloadDone
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.dnfapps.arrmatey.arr.api.model.Episode
import com.dnfapps.arrmatey.arr.api.model.EpisodeGroup
import com.dnfapps.arrmatey.compose.TabItem
import com.dnfapps.arrmatey.navigation.ArrScreen
import com.dnfapps.arrmatey.navigation.NavigationManager
import com.dnfapps.arrmatey.shared.MR
import com.dnfapps.arrmatey.ui.components.PosterItem
import com.dnfapps.arrmatey.ui.theme.ArrBlue
import com.dnfapps.arrmatey.utils.format
import com.dnfapps.arrmatey.utils.mokoString
import org.koin.compose.koinInject
import kotlin.time.ExperimentalTime

@OptIn(ExperimentalTime::class)
@Composable
fun EpisodeCalendarItem(
    episode: Episode,
    additional: List<Episode> = emptyList()
) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp),
        colors = CardDefaults.cardColors(
            containerColor = ArrBlue
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 12.dp, horizontal = 16.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            episode.series?.let { series ->
                PosterItem(item = series, Modifier.width(50.dp))
            }
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = episode.series?.title ?: mokoString(MR.strings.unknown),
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Medium
                )
                Text(
                    text = "S${episode.seasonNumber}E${episode.episodeNumber} • ${episode.title}",
                    style = MaterialTheme.typography.bodyMedium
                )

                Row(
                    modifier = Modifier.padding(top = 2.dp),
                    horizontalArrangement = Arrangement.spacedBy(8.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    episode.airDateUtc?.format("HH:mm")?.let { airTime ->
                        Text(
                            text = airTime,
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onTertiaryContainer
                        )
                    }
                    if (episode.seasonNumber == 1 && episode.episodeNumber == 1) {
                        Surface(
                            color = MaterialTheme.colorScheme.primary,
                            shape = RoundedCornerShape(4.dp)
                        ) {
                            Text(
                                text = mokoString(MR.strings.premier),
                                modifier = Modifier.padding(horizontal = 6.dp, vertical = 2.dp),
                                style = MaterialTheme.typography.labelSmall,
                                color = MaterialTheme.colorScheme.onPrimary
                            )
                        }
                    }
                    episode.finaleType?.let { finaleType ->
                        Surface(
                            color = MaterialTheme.colorScheme.primary,
                            shape = RoundedCornerShape(4.dp)
                        ) {
                            Text(
                                text = mokoString(finaleType.resource),
                                modifier = Modifier.padding(horizontal = 6.dp, vertical = 2.dp),
                                style = MaterialTheme.typography.labelSmall,
                                color = MaterialTheme.colorScheme.onPrimary
                            )
                        }
                    }

                    if (additional.isNotEmpty()) {
                        Text(
                            text = mokoString(MR.strings.additional_episodes_count, additional.size),
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onTertiaryContainer
                        )
                    }
                }
            }

            val statusIcon = when {
                episode.hasFile -> Icons.Default.FileDownloadDone
                !episode.monitored -> Icons.Default.BookmarkBorder
                !episode.hasAired -> Icons.Default.AccessTimeFilled
                episode.monitored -> Icons.Default.Bookmark
                else -> null
            }
            statusIcon?.let { icon ->
                Icon(
                    imageVector = icon,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onTertiaryContainer,
                    modifier = Modifier.size(18.dp)
                )
            }
        }
    }
}